api_3/users_get_common.py

```diff
@@ -67,75 +43,35 @@ def get(self, request):
         if user_email is None:
             raise endpoints.UnauthorizedException("Authentication unsuccessful.")
 
-        # this checks the controlled-access google group
-        am_dbgap_authorized = is_dbgap_authorized(user_email)
-
-        if not am_dbgap_authorized:
+        das = DatasetAccessSupportFactory.from_webapp_django_settings()
+        authorized_datasets = das.get_datasets_for_era_login(user_email)
+        all_datasets = das.get_all_datasets_and_google_groups()
+        
+        authorized = False
+        allowed = False
+        for dataset in authorized_datasets:
+            ad = AuthorizedDataset.objects.filter(whitelist_id=dataset.dataset_id)
+            if program in ad.name:
+                authorized = True
+                allowed = True
+        if not allowed:
```

Comment:
This part seems to negate the previous part--it's checking the program's presence in all datasets and setting allowed to true. Any given program should be found in all datasets, so what is being checked here?